sycl/test/basic_tests/host_image_accessor_read.cpp

```diff
@@ -26,50 +26,62 @@ int foo(float *image_data) {
 
     cl::sycl::range<2> pitch = Image_1.get_pitch();
 
-    auto host_image_acc =
-        Image_1.template get_access<cl::sycl::float4,
-                                    cl::sycl::access::mode::read>();
-
-    auto Sampler = cl::sycl::sampler(
-        cl::sycl::coordinate_normalization_mode::unnormalized,
-        cl::sycl::addressing_mode::none, cl::sycl::filtering_mode::nearest);
     cl::sycl::cl_int4 Coords{0, 1, 2, 0};
-    // Test image read function.
-    cl::sycl::cl_float4 Ret_data = host_image_acc.read(Coords);
-    assert((float)Ret_data.x() == 85);
-    assert((float)Ret_data.y() == 86);
-    assert((float)Ret_data.z() == 87);
-    assert((float)Ret_data.w() == 88);
+    {
+      auto host_image_acc =
+          Image_1.template get_access<cl::sycl::float4,
+                                      cl::sycl::access::mode::read>();
+
+      auto Sampler = cl::sycl::sampler(
+          cl::sycl::coordinate_normalization_mode::unnormalized,
+          cl::sycl::addressing_mode::none, cl::sycl::filtering_mode::nearest);
+      // Test image read function.
+      cl::sycl::cl_float4 Ret_data = host_image_acc.read(Coords);
+      assert((float)Ret_data.x() == 85);
+      assert((float)Ret_data.y() == 86);
+      assert((float)Ret_data.z() == 87);
+      assert((float)Ret_data.w() == 88);
+
+      // Test image read with sampler.
+      cl::sycl::cl_float4 Ret_data2 = host_image_acc.read(Coords, Sampler);
+      assert((float)Ret_data2.x() == 85);
+      assert((float)Ret_data2.y() == 86);
+      assert((float)Ret_data2.z() == 87);
+      assert((float)Ret_data2.w() == 88);
+    }
 
-    // Test image read with sampler.
-    cl::sycl::cl_float4 Ret_data2 = host_image_acc.read(Coords, Sampler);
-    assert((float)Ret_data2.x() == 85);
-    assert((float)Ret_data2.y() == 86);
-    assert((float)Ret_data2.z() == 87);
-    assert((float)Ret_data2.w() == 88);
+    {
+      auto host_image_acc2 =
```

Comment:
Nitpick: with the addition of scopes we can probably do away with the numbering of these local variables across different scopes (host_image_acc2, Sampler2).

Reply:
Done.